### ISSUE TITLE
Exclude Intent Definition files from linting

### ISSUE THREAD

**Issue by lordcodes** [tuist/tuist-plugin-lint#11]
Currently, when the plugin extracts the sources from targets, this includes '.intentdefinition' files as well as Swift files. It then lints these as if they contain Swift code when they are instead an XML file. This causes various warnings, most often due to it thinking '{}' characters are closures, when they are instead used for string interpolation.

With the regular SwiftLint binary this is handled via the '.swiftlint' file where you can exclude files or file paths from linting. Currently, there doesn't seem to be a way to customise this via the plugin.

Firstly, we can exclude this file type from linting, or perhaps we could exclude all non-Swift files? I believe SwiftLint only works on Swift sources so we could exclude all non '.swift' files maybe?

**Comment by danyf90** [comment]
I agree we should exclude all non swift file, but I would rather say it's a problem of swiftlint rather than tuist 🤔

**Comment by lordcodes** [comment]
I did wonder this myself. I have to specify various things to be excluded when I use a .swiftlint config locally and run using SwiftLint binary. However, this is due to 'Swift' files appearing in some places like dependencies or external tools. It is true however, that I don't have to exclude other file types so maybe it is true that SwiftLint itself is incorrectly thinking they are Swift files. I will have a look into it and then report there.

In the short-term it means I cannot use this plugin as it causes errors when ran on my project and so causes SwiftLint to fail. I guess I could run in non-strict mode, however, I would then not catch other warnings.

The other issue I raised #12, if that just concerns reading the excluded files from there, then we can handle it that way until the issue is fixed within SwiftLint.
